olp-cpp-sdk-dataservice-read/src/repositories/CatalogCacheRepository.cpp

```diff
@@ -111,6 +118,31 @@ boost::optional<model::VersionResponse> CatalogCacheRepository::GetVersion() {
   return boost::any_cast<model::VersionResponse>(cached_version);
 }
 
+void CatalogCacheRepository::PutVersionInfos(
+    std::int64_t start, std::int64_t end, const model::VersionInfos& versions) {
+  std::string hrn(hrn_.ToCatalogHRNString());
+  OLP_SDK_LOG_DEBUG_F(kLogTag, "PutVersionInfos -> '%s'", hrn.c_str());
+
+  cache_->Put(VersionInfosKey(hrn, start, end), versions,
+              [&]() { return olp::serializer::serialize(versions); },
+              kCatalogVersionExpiryTime);
+}
+boost::optional<model::VersionInfos> CatalogCacheRepository::GetVersionInfos(
+    std::int64_t start, std::int64_t end) {
+  std::string hrn(hrn_.ToCatalogHRNString());
+  auto key = VersionInfosKey(hrn, start, end);
+  OLP_SDK_LOG_DEBUG_F(kLogTag, "GetVersionInfos -> '%s'", key.c_str());
+
+  auto cached_versions = cache_->Get(key, [](const std::string& value) {
+    return parser::parse<model::VersionInfos>(value);
+  });
+
+  if (cached_versions.empty()) {
+    return boost::none;
+  }
+  return boost::any_cast<model::VersionInfos>(cached_versions);
+}
```

Comment:
I would prefer if we implement all new cache repositories using the non decoder/encoder versions of Put/Get and convert themselves here in CatalogCacheRepository from JSON to char* and back.

```cpp
bool Put(const std::string& key, const ValueTypePtr value, time_t expiry = kDefaultExpiry);
ValueTypePtr Get(const std::string& key)
```